File: 2024/julia/1.jl
input = readlines("input/1")

data = reduce(hcat, parse.(Int, split(line)) for line in input)
left, right = sort.(eachrow(data))

part1 = abs.(left .- right) |> sum
part2 = sum(left) do x
    x * count(==(x), right)
end

println("Part 1: ", part1)
println("Part 2: ", part2)

File: 2024/jsgolf/1.js
l=[]
r=[]
o=0
require('fs').readFileSync('1','utf8').trim().split`\n`.map(x=>{[a,b]=x.split`   `;l.push(a|0);r.push(b|0)})
l.sort();r.sort()
l.map((a,i)=>o+=Math.abs(a-r[i]))
console.log(o)